### PR TITLE
Disallow quotes in file-level directives

### PR DESCRIPTION
**Pull request by jjonescz**
### Description

This blocks usage of quotes `"` inside `#:` file-level directives, so we can later support quoted directives (https://github.com/dotnet/sdk/issues/49367) without a breaking change. This also improves the error recovery experience if users try to use quotes now thinking that's a supported syntax.

### Customer impact

If customers have quotes in their file-level directives (which seems unlikely, there is no known use case for that), they will now see errors (and would need to move these directives outside the file-based app into a Directory.Build.props file for example until we support quoted directives - this is similar to other special characters we currently don't support in unquoted directives, like trailing whitespace).

### Regression

N/A.

### Risk

Low, changing a new feature in .NET 10, disallowing an edge case that is unlikely to be relied upon. Will avoid a future breaking change.

---

Breaking change doc: https://github.com/dotnet/docs/issues/48916

### CODE REVIEW

File: test/dotnet.Tests/CommandTests/Project/Convert/DotnetProjectConvertTests.cs [hunk 1]
```diff
@@ -1561,25 +1577,43 @@ private static void Convert(string inputCSharp, out string actualProject, out st
...
+
+    private static void VerifyErrors(ImmutableArray<SimpleDiagnostic>.Builder? actual, IEnumerable<(int LineNumber, string Message)>? expected)
+    {
+        if (actual is null)
+        {
+            Assert.Null(expected);
+        }
+        else if (expected is null)
+        {
+            Assert.Null(actual);
```
[the remark this answers is not in the document]
Comment (jjonescz) [reply test/dotnet.Tests/CommandTests/Project/Convert/DotnetProjectConvertTests.cs:1621]: You're right this is basically equivalent to `Assert.Fail` but that's intentional. It also helps with nullable analysis seeing neither of these is `null` in the `else` below.

File: src/Cli/dotnet/Commands/xlf/CliCommandStrings.fr.xlf [hunk 2]
```diff
@@ -2552,6 +2552,11 @@ The default is to publish a framework-dependent application.</source>
 La valeur par défaut est de publier une application dépendante du framework.</target>
         <note />
       </trans-unit>
+      <trans-unit id="QuoteInDirective">
+        <source>Directives currently cannot contain quotes (").</source>
+        <target state="new">Directives currently cannot contain quotes (").</target>
```
[the remark this answers is not in the document]
Comment (jjonescz) [reply src/Cli/dotnet/Commands/xlf/CliCommandStrings.fr.xlf]: These are automatically generated edits.

**Comment by jjonescz** [comment]
@RikkiGibson @333fred @MiYanni for reviews, thanks

**Review by 333fred — APPROVED** [review]

**Comment by RikkiGibson** [comment]
taking a look shortly.

**Review by RikkiGibson — COMMENTED** [review]
It would be good to also demonstrate what happens when user tries to escape the quote, e.g. `#:property Name=\"Value\"`.

**Review by RikkiGibson — APPROVED** [review]
LGTM aside from a test suggestion.

File: src/Cli/dotnet/Commands/CliCommandStrings.resx [hunk 3]
```diff
@@ -1619,6 +1619,9 @@ Tool '{1}' (version '{2}') was successfully installed. Entry is added to the man
     <value>Duplicate directives are not supported: {0}</value>
     <comment>{0} is the directive type and name.</comment>
   </data>
+  <data name="QuoteInDirective" xml:space="preserve">
+    <value>Directives currently cannot contain quotes (").</value>
```
Comment (MiYanni) [inline src/Cli/dotnet/Commands/CliCommandStrings.resx]: So, is this specific to double quotes `"` or does that also account for single quotes `'`? Might make sense to be a bit more specific.

**Review by MiYanni — APPROVED** [review]

File: src/Cli/dotnet/Commands/CliCommandStrings.resx (hunk 3 above)
Comment (jjonescz) [reply src/Cli/dotnet/Commands/CliCommandStrings.resx]: Currently this is specific to double quotes, I will clarify, thanks. We could disallow single quotes too, I didn't think about that, but it also seems unnecessary (we likely won't want to support single-quoted directive values anyway, only double-quoted, I imagine, same as C# strings).
